### ISSUE TITLE
auto-wrap-observer is breaking Fast Refresh

### ISSUE THREAD

**Issue by zhigang1992** [noru/use-mobx#2]
This project is really great!

Unfortunately with a vite swc react project, fast refresh would stop working as soon as we turn on auto-wrap-observer.

Any thoughts on why that might be?

**Comment by noru** [comment]
Would you provide more info on this? or a working example. I would be happy to look at it.

**Comment by zhigang1992** [comment]
Thank you!

https://github.com/zhigang1992/mobx-fast-refresh

Here is the reproduce repo.

`pnpm create vite mobx-fast-refresh`

And then adding `use-mobx-observable` to the `vite.config.ts`

As soon as the `jsxImportSource` is added, the fast refresh would stop working

**Comment by noru** [comment]
TLDR; 
`Don't use it. I'll remove it from Doc.`

I recalled the issue. 
This `auto-wrap-observer` was introduced in a beta version and immediately after release I found several limitations of this approach in my own project. 
To counter that, I need to impose some measure (like using a decorator or what not), which somehow against the intention ("auto wrap").
I reverted the use case in my project and somehow forgot to update the document, which causes a lot confusion, really sorry about that!

Thanks for reporting.
